llvmlite/ir/_utils.py

```diff
@@ -53,3 +53,20 @@ def get_reference(self):
         except AttributeError:
             s = self.__cached_refstr = self._get_reference()
             return s
+
+
+class _HasMetadata(object):
```

Comment:
As a documentation to ourselves, to know why this exists ;-)
As for the `set_metadata` method, I'm assuming it is a public API?